### PR TITLE
Test harness with jest

### PR DESCRIPTION
**Pull request by awmartin**
Proposing to create a node-based test harness using the popular [jest](https://jestjs.io/) library. This could be used to enable a few workflows:

* First, the typical value propositions around testing: ensuring changes don't break expected behavior, stability, etc.
* Enabling some basic test-driven-development. For example, a reported bug can be confirmed with a failing test, then a branch created to fix it, code to make the test pass, then a PR to submit. This could include adding `*.ifc` files representing import challenges.
* In the near future, running the test suite as part of commit flow or automated build pipelines.

This PR adds `jest` to the development dependencies and a new build action, `test`, to be executed with `npm run test`. Individual tests can be singled out with jest's features (regex, filename, etc.). In the future, we could add `test:unit`, `test:e2e`, etc. commands to single out test types.

It also proposes a fix to a bug uncovered with the included test.

### CODE REVIEW

**Comment by awmartin** [comment]
Going to convert to typescript as per @tomvandig's request.

File: src/web-ifc-api.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {IfcElements} from "./ifc2x4";
 let WebIFCWasm;
 
 //@ts-ignore
-if (self.crossOriginIsolated)
+if (typeof self !== 'undefined' && self.crossOriginIsolated)
```
Comment (tomvandig) [inline src/web-ifc-api.ts:10]: nice!

**Comment by awmartin** [comment]
`.gitignore` excludes .ifc files, but examples/example.ifc seems to have already been included. Keeping test artifacts, particularly representative, small, IFC files, close to the functional tests might be convenient. One way to do this would be to reference `test-ifc-files` as a git submodule.

**Comment by awmartin** [comment]
I think this is a good start to testing on the Typescript/Javascript side of the interface. I'd be curious to integrate this command `npm run test` (or `npm test`) with the CPP side as well. I see there's the start to those tests already.

The process uncovered a couple of bugs already, which I took the liberty to fix. :-)

One cavaet we should document once more tests are added is this requires a full build in the dist folder.

**Pull request merged**